#! /bin/sh /usr/share/dpatch/dpatch-run
## codecs_Makefile.dpatch by Mark Purcell <msp@debian.org>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: Use installed libgsm if availble

@DPATCH@
diff -urNad asterisk-1.2.0.dfsg~/codecs/codec_gsm.c asterisk-1.2.0.dfsg/codecs/codec_gsm.c
--- asterisk-1.2.0.dfsg~/codecs/codec_gsm.c	2005-11-06 15:09:46.000000000 +0000
+++ asterisk-1.2.0.dfsg/codecs/codec_gsm.c	2005-11-21 19:52:33.000000000 +0000
@@ -45,7 +45,7 @@
 #include "asterisk/logger.h"
 #include "asterisk/channel.h"
 
-#include "gsm/inc/gsm.h"
+#include <gsm/gsm.h>
 #include "../formats/msgsm.h"
 
 /* Sample frame data */
diff -urNad asterisk-1.2.0.dfsg~/codecs/Makefile asterisk-1.2.0.dfsg/codecs/Makefile
--- asterisk-1.2.0.dfsg~/codecs/Makefile	2005-11-08 04:13:18.000000000 +0000
+++ asterisk-1.2.0.dfsg/codecs/Makefile	2005-11-21 19:53:31.000000000 +0000
@@ -60,8 +60,9 @@
 endif
 
 
-LIBGSM=gsm/lib/libgsm.a
-LIBGSMT=gsm/lib/libgsm.a
+LIBGSM=-lgsm
+# We don't build libgsm but use the system's one.
+LIBGSMT=
 LIBLPC10=lpc10/liblpc10.a
 
 ifeq ($(findstring BSD,${OSARCH}),BSD)
diff -urNad asterisk-1.2.0.dfsg~/formats/Makefile asterisk-1.2.0.dfsg/formats/Makefile
--- asterisk-1.2.0.dfsg~/formats/Makefile	2005-11-01 21:53:30.000000000 +0000
+++ asterisk-1.2.0.dfsg/formats/Makefile	2005-11-21 19:54:10.000000000 +0000
@@ -33,8 +33,6 @@
   CFLAGS+=-I$(CROSS_COMPILE_TARGET)/usr/local/include -L$(CROSS_COMPILE_TARGET)/usr/local/lib
 endif
 
-GSMLIB=../codecs/gsm/lib/libgsm.a
-
 ifeq (${OSARCH},CYGWIN)
 CYGSOLINK=-Wl,--out-implib=lib$@.a -Wl,--export-all-symbols
 CYGSOLIB=-L.. -L. -lasterisk.dll
@@ -53,6 +51,12 @@
   include .depend
 endif
 
+format_gsm.so : format_gsm.o
+	$(CC) $(SOLINK) -o $@ $< -lgsm
+
+format_wav_gsm.so : format_wav_gsm.o
+	$(CC) $(SOLINK) -o $@ $< -lgsm
+
 format_mp3.so : format_mp3.o
 	$(CC) $(SOLINK) -o $@ ${CYGSOLINK} $< ${CYGSOLIB} -lm
 
